var/spack/repos/builtin/packages/py-dask/package.py

```diff
@@ -149,13 +156,16 @@ class PyDask(PythonPackage):
     depends_on("py-distributed@1.22:2021.8.0", type=("build", "run"), when="@0.18.0: +distributed")
     depends_on("py-distributed@2.0:2021.8.0", type=("build", "run"), when="@2.0.0: +distributed")
```

Comment:
You'll also need to add upper bounds to all `when` args in older distributed deps, otherwise these constraints will also apply to the latest version.

Reply:
How can I test that I did this correctly?

Reply:
You can run `spack spec py-dask@X.Y.Z+distributed` for various versions and make sure it concretizes correctly. I'll also spot check it when I'm back at my computer.